pallets/rolldown/src/lib.rs

```diff
@@ -311,6 +311,50 @@ pub mod pallet {
 			Ok(().into())
 		}
 
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::DbWeight::get().reads_writes(1, 1).saturating_add(Weight::from_parts(40_000_000, 0)))]
+		pub fn withdraw(
+			origin: OriginFor<T>,
+			tokenAddress: [u8; 20],
+			amount: u128,
+		) -> DispatchResultWithPostInfo {
+			let account = ensure_signed(origin)?;
+
+			let eth_token_address: [u8; 20] = array_bytes::hex2array::<_, 20>(tokenAddress.clone())
+				.or(Err("Cannot convert address"))?;
+			let eth_asset = L1Asset::Ethereum(eth_token_address);
+			let asset_id = T::AssetRegistryProvider::get_l1_asset_id(eth_asset.clone())
+				.ok_or("L1AssetNotFound")?;
+
+			// fail will occur if user has not enough balance
+			<T as Config>::Tokens::ensure_can_withdraw(
+				asset_id.into(),
+				&account,
+				amount.try_into().map_err(|_| "u128 to Balance failed")?,
```

Comment:
please use regular substrate Errors and not string

Reply:
done